components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/util/Utils.java

```diff
@@ -382,4 +382,32 @@ public static Policy getSecurityPolicy() {
         return PolicyEngine.getPolicy(new ByteArrayInputStream(policyString.getBytes()));
 
     }
+
+    /**
+     * check the case sensitivity of user store
+     *
+     * @param username
+     * @param tenantId
+     * @return
+     */
+    public static boolean isUsernameCaseSensitive(String username, int tenantId) {
+
+        boolean isUsernameCaseSensitive = true;
+        try {
+
+            org.wso2.carbon.user.core.UserStoreManager userStoreManager = (org.wso2.carbon.user.core.UserStoreManager) IdentityMgtServiceComponent.getRealmService()
+                    .getTenantUserRealm(tenantId).getUserStoreManager();
+            org.wso2.carbon.user.core.UserStoreManager UserAvailableUserStoreManager = userStoreManager.getSecondaryUserStoreManager
+                    (UserCoreUtil.extractDomainFromName(username));
+            String caseInsensitiveUsername = UserAvailableUserStoreManager.getRealmConfiguration().getUserStoreProperty("CaseInsensitiveUsername");
```

Comment:
For "CaseInsensitiveUsername", use a constant. you can reuse one constant for the entire component.